include/contractor/graph_contractor_adaptors.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef OSRM_CONTRACTOR_GRAPH_CONTRACTION_ADAPTORS_HPP_
+#define OSRM_CONTRACTOR_GRAPH_CONTRACTION_ADAPTORS_HPP_
+
+#include "contractor/contractor_graph.hpp"
+#include "util/log.hpp"
+
+#include <vector>
+
+namespace osrm
+{
+namespace contractor
+{
+
+// Make sure to move in the input edge list!
+template <typename InputEdgeContainer>
+std::vector<ContractorEdge> ContractorEdgeInputAdaptor(InputEdgeContainer input_edge_list)
+{
+    std::vector<ContractorEdge> edges;
+    edges.reserve(input_edge_list.size() * 2);
+
+    for (const auto &input_edge : input_edge_list)
+    {
+#ifndef NDEBUG
+        if (static_cast<unsigned int>(std::max(input_edge.weight, 1)) > 24 * 60 * 60 * 10)
```

Comment:
Not scope of this PR just a though: Can we replace this by a real overflow check?